test/asset/raw_datasets.jsonl

```diff
@@ -1,7 +1,7 @@
 {"dataset_name": "IMDB", "split": "train", "NUM_LINES": 25000, "MD5": "7c2ac02c03563afcf9b574c7e56c153a", "URL": "http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz", "first_line": "7d45c2c4463561748d304b9e53ea762a"}
 {"dataset_name": "IMDB", "split": "test", "NUM_LINES": 25000, "MD5": "7c2ac02c03563afcf9b574c7e56c153a", "URL": "http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz", "first_line": "80dd1b651d74220e41f7c7c4d0be1d7c"}
-{"dataset_name": "AG_NEWS", "split": "train", "NUM_LINES": 120000, "MD5": "b1a00f826fdfbd249f79597b59e1dc12", "URL": "https://raw.githubusercontent.com/mhjabreel/CharCnn_Keras/master/data/ag_news_csv/train.csv", "first_line": "b34631d550696857784c780e75d1019b"}
-{"dataset_name": "AG_NEWS", "split": "test", "NUM_LINES": 7600, "MD5": "d52ea96a97a2d943681189a97654912d", "URL": "https://raw.githubusercontent.com/mhjabreel/CharCnn_Keras/master/data/ag_news_csv/test.csv", "first_line": "4ec431dd3276af3ddaa26a18687e134f"}
+{"dataset_name": "AG_NEWS", "split": "train", "NUM_LINES": 120000, "MD5": "b1a00f826fdfbd249f79597b59e1dc12", "URL": "https://raw.githubusercontent.com/mhjabreel/CharCnn_Keras/master/data/ag_news_csv/train.csv", "first_line": "b34631d550696857784c780e75d1019b", "offset": 11010, "offset_line": "b7923bb47f29e45f94cfc02c2c92e7cd"}
```

Comment:
I'd suggest to double the lines and make offset=0 the default when not specified for all other test. So you'll have one line with offset 0 and then another with an explicit offset set instead of additional flags. Otherwise as we add flags we might end up with offset_line, offset and chunk_line, chunk, etc.

Reply:
I think this much better to avoid testing complexity @salilsdesai

Reply:
@cpuhrsch @parmeet. I'm not sure that I entirely understand what you mean. Does this mean that I add offset=0 and offset_line=... to each line in the original file, then for AG_NEWS train and AG_NEWS split, I add duplicate lines but with the non-zero offset and corresponding offset line?

Reply:
yes, I think this would be the idea. And in our unit test code, since we do not have offsets argument available for other datasets, we may just ignore the test cases with non-zero offsets for datasets other than AG_NEWS.

Reply:
> we may just ignore the test cases with non-zero offsets for datasets other than AG_NEWS

Do you mean "ignore the test cases with **zero** offsets"? Since none of the test cases will have non-zero offsets except AG_NEWS.

Reply:
After giving it a second thought, we can just include the lines with non-zero offsets for the datasets that supports offset (AG_NEWS for now) and rest of the lines with offset=0. The unit test [here](https://github.com/pytorch/text/blob/master/test/data/test_builtin_datasets.py#L58) will iterate over each line provided in the raw_datasets.jsonl file. When offset=0, we simply run the test without passing the offset to datasets (this is valid behaviour for datasets both that support and that doesn't support offset), and when offset is non-zero (only AG_NEWS for now), we call the dataset with given offset.

Reply:
Yeah I think that makes more sense too. The pull request is updated with this now.